ocaml/xapi/xapi_vm.ml

```diff
@@ -1444,6 +1444,31 @@ let set_appliance ~__context ~self ~value =
   (* Update the VM's allowed operations - this will update the new appliance's operations, if valid. *)
   update_allowed_operations ~__context ~self
 
+let set_groups ~__context ~self ~value =
+  let assert_only_one_placement_group groups singleton_placements =
```

Comment:
I'd really just do something like
```
if (List.length value > 1) then raise ... 
```
to restrict to one group per VM in general.